fl-server/src/auth.rs

```diff
@@ -70,17 +70,17 @@ pub async fn sign_in_handler(
         ));
     }
 
-    let token = encode_jwt(user.username, cfg.jwt_secret, cfg.jwt_expire_hours)
+    let token = encode_jwt(user.username.clone(), cfg.jwt_secret, cfg.jwt_expire_hours)
         .map_err(|_| ResponseError::InternalServerError)?;
 
     Ok(ResponseResult::SignedIn(SignInResponse {
         access_token: token,
     }))
 }
 
-pub fn get_user_by_username(users: Vec<User>, username: &str) -> Option<User> {
+pub fn get_user_by_username<'a>(users: &'a Vec<User>, username: &str) -> Option<&'a User> {
```

Comment:
```suggest
pub fn get_user_by_username(users: &[User], username: &str) -> Option<&'a User> {
```
This can be called with a `&Vec<T>` normally because it can deref as a slice